File: api-typescript/samples/server/src/hosted.ts
import { SessionsApi } from "@trinsic/api";
import path from "path";
import {Express} from "express"

export function hostedRoutes(app: Express, sessionsApi: SessionsApi) {
    app.get("/hosted", (req: any, res: any) => {
        res.sendFile(
            path.join(__dirname, "../../../../ui-web/samples/dist/hosted.html")
        );
    });

    app.get("/hosted-launch/:provider", async (req: any, res: any) => {
        const provider = req.params.provider;
        const redirectUrl = req.query.redirectUrl;

        const result = await sessionsApi.createHostedProviderSession({
            provider: provider,
            redirectUrl: redirectUrl
        });
        res.redirect(result.launchUrl);
    });
}

File: api-go/samples/server/hosted.go
package main

import (
	"github.com/gofiber/fiber/v2"
	"os"
	trinsic_api "github.com/trinsic-id/sdk-go-api/v2"
)

func HostedRoutes(app *fiber.App, api *trinsic_api.APIClient) {
	app.Get("/hosted", func(c *fiber.Ctx) error {
		return c.Redirect("/hosted.html")
	})

	app.Get("/hosted-launch/:provider", func(c *fiber.Ctx) error {
		provider := c.Params("provider")
		redirectUrl := c.Query("redirectUrl")
		verificationProfileId := os.Getenv("TRINSIC_VERIFICATION_PROFILE_ID")
		// Create request with provider and nullable redirectUrl
		request := trinsic_api.CreateHostedProviderSessionRequest{
			RedirectUrl: redirectUrl,
			Provider:    provider,
			VerificationProfileId: verificationProfileId,
		}

		req := api.SessionsAPI.CreateHostedProviderSession(c.Context()).CreateHostedProviderSessionRequest(request)
		data, _, err := req.Execute()

		if err != nil {
			return err
		}

		return c.Redirect(data.GetLaunchUrl())
	})
}
